helios-system-tests/src/main/java/com/spotify/helios/system/SystemTestBase.java

```diff
@@ -602,7 +615,34 @@ protected AgentMain startDefaultAgent(final String host, final String... args)
         dockerPortRange.upperEndpoint()
     );
     argsList.addAll(asList(args));
-    return startAgent(argsList.toArray(new String[argsList.size()]));
+    return startAgentWithArgs(argsList);
+  }
+
+  protected AgentMain startAgentWithArgs(final List<String> args) throws Exception {
+    final AgentMain agentMain = startAgent(args.toArray(new String[args.size()]));
+    waitForAgentToBeFullyUp();
+    return agentMain;
+  }
+
+  protected void waitForAgentToBeFullyUp() throws Exception {
```

Comment:
this looks identical to `waitForMasterToBeFullUp()`, can you extract the callable so that both functions can use the same code?